ui/components/src/ui-components/filtering/FilteringInput.tsx

```diff
@@ -20,41 +21,83 @@ import { ResultSelector, ResultSelectorProps } from "./ResultSelector";
 interface FilteringInputState {
   /** A string which will be used for search */
   searchText: string;
-  /** @internal */
-  context: InputContext;
+  /**
+   *  Tells the component if the search was started.
+   *  Gets reset to false if search is canceled/cleared or searchText is changed.
+  */
+  searchStarted: boolean;
+  /* Used for resetting the state of [[ResultSelector]] component */
+  resultSelectorKey: number;
+  /**
+   *  Parameter used to remember `props.filteringInProgress`.
+   *  Used in getDerivedStateFromProps to compare previous `props.filteringInProgress`
+   *  with current `props.filteringInProgress`.
+   *  Used for supporting deprecated usage of [[FilteringInput]] component
+   */
+  filteringInProgress?: boolean;
```

Comment:
Maybe rename to `prevFilteringInProgress` or `wasFilteringInProgress`?

Reply:
Renamed